op-challenger/game/fault/trace/super/provider.go

```diff
@@ -76,13 +76,21 @@ func (s *SuperTraceProvider) GetPreimageBytes(ctx context.Context, pos types.Pos
 		if err != nil {
 			return nil, fmt.Errorf("failed to retrieve super root at timestamp %v: %w", timestamp, err)
 		}
+		if root.CrossSafeDerivedFrom.Number > s.l1Head.Number {
+			return InvalidTransition, nil
```

Comment:
I'm not sure I understand the step = 0 case. Don't we assume the superroot is agreed, so this shouldn't happen?

Reply:
The trace provider only deals with correct inputs - there's no agreed or disagreed here.  When step is 0 it means the correct claim will be a super root (not a transition state) and so we load the correct root from the supervisor, then we check if the super root was safe at the game's L1 head which is this new check.

We actually also need to handle the case where the supervisor can't return a super root for the timestamp at all because the timestamp is in the future. Logged https://github.com/ethereum-optimism/optimism/issues/14077 for that. We get an error from `optimism_outputAtBlock` in that case, but not sure what op-supervisor does and we'll need to detect it.

Reply:
Got it. Thanks.